tidysic/ordering.py

```diff
@@ -0,0 +1,24 @@
+from tidysic.tag import Tag
+from tidysic.formatted_string import FormattedString
+
+
+class OrderingStep:
+
+    def __init__(self, tag: Tag, formatted_string: FormattedString):
+        self.tag = tag
+        self.format = formatted_string
+
+
+class Ordering:
```

Comment:
I've never seen a recursive class like this. A simpler design with similar functionalities can be obtained with a function:

```python
def ordering(steps: List[OrderingStep] = []):
    if len(steps) > 1:
        ordering(steps[1:])
    return steps
```